modules/ROOT/pages/auth.adoc

```diff
@@ -0,0 +1,187 @@
+----
++
+[TIP]
+====
+The names of the files used in this section match the default values in the Helm chart's `values.yaml` file. If you used different names, then you will have to update the corresponding values.
+====
+
+. Store those keys as secrets.
++
+----
+kubectl create secret generic token-private-key \
+ --from-file=my-private.key \
+ --namespace pulsar
+----
++
+----
+kubectl create secret generic token-public-key \
+ --from-file=my-public.key \
+ --namespace pulsar
+----
+
+. Using those keys, generate tokens for the following subjects (roles): 
++
+* `admin`
+* `superuser`
+* `proxy`
+* `websocket` (only required if using the standalone WebSocket proxy)
++
+----
+bin/pulsar tokens create --private-key file:///pulsar/token-private-key/my-private.key --subject <subject>
+----
+
+. Add each token as a secret:
++
+----
+kubectl create secret generic token-<subject> \
+ --from-file=<subject>.jwt \
+ --namespace pulsar
+----
+
+. Enable token-based authentication with this setting in `values.yaml`:
++
+----
+enableTokenAuth: yes
+----
+
+== TLS
+
+=== Automatically generating certificates using cert-manager
+
+To automatically generate certificates using cert-manager, see https://cert-manager.io/docs/[Cert-Manager Documentation].
+
+=== Manually configuring certificate secrets for TLS
+
+To use TLS, you must first create a certificate and store it in the secret defined by `tlsSecretName`.
+
+Create the certificate:
+
+`kubectl create secret tls <tlsSecretName> --key <keyFile> --cert <certFile>`
+
+The resulting secret will be of type `kubernetes.io/tls`. The key should *not* be in `PKCS 8` format, even though that is the format used by Pulsar. The `kubernetes.io/tls` format will be converted by the chart to `PKCS 8`. 
+
+If you have a self-signed certificate, manually specify the certificate information directly in https://github.com/datastax/pulsar-helm-chart/blob/master/examples/dev-values-keycloak-auth.yaml[values]:
+
+----
+# secrets:
+  # key: |
+  # certificate: |
+  # caCertificate: |
+----
+
+Once you have created the secrets that store the certificate info (or manually specified it in https://github.com/datastax/pulsar-helm-chart/blob/master/examples/dev-values-keycloak-auth.yaml[values]), enable TLS in the values:
+
+`enableTls: yes`
+
+== Token Authentication via Keycloak Integration
+
+DataStax created the https://github.com/datastax/pulsar-openid-connect-plugin[Pulsar OpenID Connect Authentication Plugin] to provide a more dynamic authentication option for Pulsar. This plugin integrates with Keycloak to dynamically retrieve Public Keys from a running Keycloak instance for token validation. This dynamic public key retrieval enables support for key rotation and multiple authentication/identity providers by configuring multiple allowed token issuers. It also means that token secret keys will *not* be stored in Kubernetes secrets.
```

Comment:
Nit: the plugin is supposed to integrate with any OpenID Connect compliant Identity Provider. We use Keycloak as an example, but the spec should allow us to integrate with Okta, Azure AD, and other providers. We haven't not yet tested with these providers, so there is a chance that small adjustments will need to be made.